src/java/com/synacor/zimbra/ys/contacts/YahooContactsUtil.java

```diff
@@ -327,61 +344,77 @@ public static ParsedContact parseYContact(JsonObject jsonContact, DataSource ds)
                     if(next.isJsonObject()) {
                         JsonObject fieldObj = next.getAsJsonObject();
                         if(fieldObj.has(TYPE) && fieldObj.has(VALUE)) {
-                            YahooContactsUtil.YContactFieldType type = YahooContactsUtil.YContactFieldType.valueOf(fieldObj.get(TYPE).getAsString().toLowerCase());
-                            switch (type) {
-                            case email:
-                                parseFlaggedField(fieldObj, A_email, EMAIL_FIELDS_MAP, contactFields);
-                                break;
-                            case phone:
-                                parseFlaggedField(fieldObj, A_homePhone, PHONE_FIELDS_MAP,  contactFields);
-                                break;
-                            case address:
-                                parseAddressField(fieldObj, contactFields);
-                                break;
-                            case birthday:
-                            case anniversary:
-                                Locale locale = null;
-                                if(ds != null) {
-                                    locale = ds.getAccount().getLocale();
+                            String fieldType = fieldObj.get(TYPE).getAsString().toLowerCase();
+                            YahooContactsUtil.YContactFieldType type = null;
+                            if(fieldType != null && !fieldType.isEmpty()) {
+                                try {
+                                    type = YahooContactsUtil.YContactFieldType.valueOf(fieldType);
+                                } catch(Exception e) {
+                                    ZimbraLog.extensions.warn("YahooContactsUtil cannot map Yahoo Contact field of type '%s' to any Zimbra contact field. Ignoring the field.", fieldType);
```

Comment:
I'm sure contacts has a way to store unknown data with some kind of key=value mechanism??